Liferea/Liferea

```diff
@@ -24,137 +24,168 @@
+    self.number_of_unread_msgs = 0
+    self.number_of_new_msgs = 0
+    self.show_number_of_unread_msgs = 0
+    self.show_number_of_new_msgs = 1
+
+    self.set_status_online_key = 1
+    self.set_status_offline_key = 2
+    self.refresh_key = 3
     
-    def bind_to_liferea_window(self):
-        self.icon.ControlAppli("liferea")
-
-    def build_status_menu(self):
-        status_menu = []
-        index = 1
-        for status in ["Online", "Offline"]:
-            status_menu.append({
-                'type'  : CDApplet.MENU_ENTRY,
-                'label' : status,
-                'menu'  : CDApplet.MAIN_MENU_ID,
-                'id'    : index,
-                'icon'  : os.path.abspath("./data/%s.png") % status.lower()})
-            index += 1
-        self.icon.AddMenuItems(status_menu)
-
-    # CDApplet
-
-    def begin(self):
-        self.bind_to_liferea_window()
-        if self.connect_to_liferea():
-            self.get_number_of_unread_msgs()
-            self.get_number_of_new_msgs()
-            self.show_icon_information()
-
-    def get_config(self, keyfile):
-        self.config_info = keyfile.getint('Configuration', 'icon_info')                 # get the source of quotations
-
-    def reload(self):
-        self.show_icon_information()
-
-    # Callbacks
-
-    def on_click(self, param):
-        if self.liferea:
-            self.switch_window_focus()
-
-    # Probably the user read something when he focus Liferea, therefore,
-    # it is useful to check again and refresh the number of unread, and new items
-    # I will avoid polling to keep checking this variables
-    def on_change_focus(self, has_focus):
-        self.get_number_of_unread_msgs()
-        self.get_number_of_new_msgs()
-        self.show_icon_information()
-
-    def on_drop_data(self, text):
-        self.subscribe(text)
-
-    def on_middle_click(self):
-        self.icon.PopupDialog ({'message':'Enter the feed URL', 'buttons':'ok;cancel'}, {'widget-type':'text-entry'})
-
-    def on_answer_dialog(self, key, content):
-        if (key == 0 or key == -1) and content:                                         # Ok or Enter
-            self.subscribe(content)
-
-    def on_build_menu(self):
-        self.build_status_menu()
-
-    def on_menu_select(self, selected_menu):
-        if selected_menu == self.set_status_online_key:
-            self.set_online()
-        elif selected_menu == self.set_status_offline_key:
-            self.set_offline()
+    self.shown_disconnected_alert = False
+
+    self.dialog_active_time = 30                                                    # time in seconds that the dialog window will be active
+    CDApplet.__init__(self)                                                         # call high-level init
+
+  def connect_to_liferea(self):
+    while not self.liferea:
+      try:
+        bus = dbus.SessionBus()
+        liferea_object = bus.get_object('org.gnome.feed.Reader', '/org/gnome/feed/Reader')
```

Comment:
Is it not better to use `bus.watch_name_owner(...)` method here?

You can have a look to the code of other applets like XChat: https://github.com/Cairo-Dock/cairo-dock-plug-ins-extras/blob/master/Xchat/Xchat#L213